### DISCUSSION TITLE
get theme colors via property from component failed

### DISCUSSION THREAD

**Discussion by minzojian** [ben-rogerson/twin.macro#764] [Q&A]
suppose i have define a simple component like this

```
const MyText=({color,children})=>{
  return <span  css={{
      color:`${theme(`colors.${color}`)}`,
    }} >{children}</span>
}
```

i try to use theme color via the component property, however, it is not working.and says:

```
✕ The theme value doesn’t look right

Try theme`colors.black` or theme('colors.black')
```

how to use theme colors or othere config in runtime?

**Comment by ben-rogerson** [comment]
Check out [this discussion](https://github.com/ben-rogerson/twin.macro/discussions/393) on this topic

**Comment by minzojian** [comment]
so if i have lots of variables, is that mean i should re-map them all ?
```
import tw from 'twin.macro'

const styles = { color1: theme`colors.xxx1`,<..lots of colors re-define...> ,color100:theme`colors.xxx100` }

const Component = ({ color = 'color1' }) => <div css={styles[color]} />
```

a little silly :(

**Reply by ben-rogerson** [reply]
it'll work well for a smaller number of colors - if you have more then perhaps try importing tailwind.config and parsing for your colors

**Comment by minzojian** [comment]
how about we use some babel plugins to export the tailwindcss theme into a global variable in js? so that can use it in runtime
